Sources/Benchmark/BenchmarkRunner.swift

```diff
@@ -18,10 +18,22 @@ public struct BenchmarkRunner {
     var reporter: BenchmarkReporter
     var results: [BenchmarkResult] = []
 
-    init(suites: [BenchmarkSuite], settings: [BenchmarkSetting], reporter: BenchmarkReporter) {
+    init(suites: [BenchmarkSuite], settings: [BenchmarkSetting]) {
         self.suites = suites
         self.settings = settings
-        self.reporter = reporter
+
+        var format: BenchmarkFormat = .console
+        for case .benchmarkFormat(let value) in settings {
+            format = value
+            break
+        }
+
+        switch format {
+        case .console:
+            self.reporter = PlainTextReporter()
+        case .json:
+            self.reporter = JSONReporter()
+        }
```

Comment:
Let's keep `blackHole` as another option here, if it's used in tests only.

Reply:
Could you say that `BlackHoleReporter` functionally equivalent to specifying a `nil` reporter? If so, I think that could allow for a simpler API overall.

Reply:
`nil` reporter sounds good to me.